### PR TITLE
Update backup validation docs for cluster restore and enterprise license

### PR DESCRIPTION
**Pull request by kathancox**
Fixes DOC-9050

PR adds notes on the following:

- Users do not need an Enterprise license to use these tools
- To run `schema-only` for full-cluster restores, the cluster must be empty.

Removed the "Cluster-level" section as this seemed to be causing some confusion, and is not strictly necessary information.

### CODE REVIEW

File: src/current/v22.2/restore.md [hunk 1]
```diff
@@ -389,8 +389,8 @@ RESTORE DATABASE bank FROM LATEST IN 's3://{bucket_name}?AWS_ACCESS_KEY_ID={key_
 {{site.data.alerts.end}}
 
 ### Restore with `AS OF SYSTEM TIME`
- 
-Running a backup with [revision history](take-backups-with-revision-history-and-restore-from-a-point-in-time.html) captures every change made within the garbage collection period leading up to and including the given timestamp, which allows you to restore to an arbitrary point-in-time within the revision history.
```
Comment (msbutler) [inline src/current/v22.2/restore.md:393]: looks like some random formatting crept into the diff. was that intentional?

**Review by msbutler — APPROVED** [review]

File: src/current/v22.2/restore.md (hunk 1 above)
Comment (kathancox) [reply src/current/v22.2/restore.md:393]: Kind of.... I now have an automated thing in my VS Code to remove any trailing spaces. I didn't have that enabled for a long time, so where I have not touched a file in a while, it's automatically removing any trailing spaces I had left in. Thanks for checking!

File: src/current/v23.2/backup-validation.md [hunk 2]
```diff
@@ -15,13 +15,17 @@ You can validate a backup of a [cluster]({% link {{ page.version.version }}/back
...
+{{site.data.alerts.end}}
+
 ## Recommendations
 
 Cockroach Labs recommends implementing the following validation plan to test your backups:
 
-1. Very frequent [`schema_only`](#validate-a-backup-is-restorable) restores: Checks your credentials, schema issues, and version compatability.  
-1. Frequent [`verify_backup_table_data`](#validate-backup-table-data-is-restorable) restores: Tests that all data files are present and uncorrupted. 
-1. Somewhat frequent ["full"]({% link {{ page.version.version }}/restore.md %}) restores: Validates a backup completely. 
+1. Very frequent [`schema_only`](#validate-a-backup-is-restorable) restores: Checks your credentials, schema issues, and version compatability.
```
Comment (florence-crl) [inline src/current/v23.2/backup-validation.md:26]: minor spelling error: compatability should be compatibility

**Review by florence-crl — APPROVED** [review]
LGTM 1 small nit

File: src/current/v23.2/backup-validation.md (hunk 2 above)
Comment (kathancox) [reply src/current/v23.2/backup-validation.md:26]: whoooops! Thanks for the catch on that Florence!

**Pull request merged**